File: services/mono/Bacera.Gateway.Core/Models/Tenant/Application.cs
using System;
using System.Collections.Generic;

namespace Bacera.Gateway;

public partial class Application
{
    public long Id { get; set; }

    public long PartyId { get; set; }

    public short Type { get; set; }

    public short Status { get; set; }

    public long ReferenceId { get; set; }

    public DateTime? ApprovedOn { get; set; }

    public long? ApprovedBy { get; set; }

    public DateTime? RejectedOn { get; set; }

    public long? RejectedBy { get; set; }

    public string? RejectedReason { get; set; }

    public DateTime CreatedOn { get; set; }

    public DateTime UpdatedOn { get; set; }

    public DateTime? CompletedOn { get; set; }

    public long? CompletedBy { get; set; }

    public string? Supplement { get; set; }

    public virtual Party Party { get; set; } = null!;
}

File: app/web/vue/src/projects/client/modules/accounts/model/Application.ts
import { ICriteria } from "@/core/models/Criteria";

export interface Application {
  id: number;
  partyId: number;
  type: number;
  approvedOn: string | null;
  approvedBy: number | null;
  rejectedOn: string | null;
  rejectedBy: number | null;
  rejectedReason: string | null;
  createdOn: string;
  updatedOn: string;
  supplement: unknown | null;
  user: unknown | null;
}

export interface ApplicationCriteria extends ICriteria {
  partyId: number | null;
  type: number | null;
  from: string | null;
  to: string | null;
  /**
   *     AwaitingApproval = 1,
   *     Approved = 2,
   *     Rejected = 3,
   */
  status: number | null;
}
